libraries/libfc/include/fc/crypto/bls_private_key.hpp

```diff
@@ -1,50 +1,227 @@
 #pragma once
+#include <ranges>
+#include <fc/crypto/common.hpp>
 #include <fc/crypto/bls_public_key.hpp>
 #include <fc/crypto/bls_signature.hpp>
+#include <fc/crypto/sha512.hpp>
 #include <fc/reflect/reflect.hpp>
-#include <fc/reflect/variant.hpp>
 
-namespace fc::crypto::blslib {
 
-   namespace config {
-      const std::string bls_private_key_prefix = "PVT_BLS_";
+/**
+ * @brief Namespace for BLS cryptographic functionality
+ */
+namespace fc::crypto::bls {
+
+/**
+ * @brief Constants used in BLS private key operations
+ */
+namespace constants {
+/** @brief Prefix used for BLS private key string representation */
+const std::string bls_private_key_prefix = "PVT_BLS_";
+};
+
+/**
+ * @brief Class representing a BLS private key
+ */
+class private_key {
+public:
+   /** @brief Default constructor */
+   private_key() = default;
+   /** @brief Move constructor */
+   private_key(private_key&&) = default;
+   /** @brief Copy constructor */
+   private_key(const private_key&) = default;
+   /** @brief Constructs a private key from a secret */
+   explicit private_key(const private_key_secret& sk) : _sk(sk) {}
+
+   /** 
+    * @brief Constructs a private key from a seed
+    * @param seed Byte span containing the seed data
+    */
+   explicit private_key(std::span<const uint8_t> seed) {
+      auto key_data = bls12_381::secret_key(seed);
+      _sk = private_key_secret(reinterpret_cast<const char*>(key_data.data()), sizeof(key_data));
```

Comment:
If `private_key_secret` is changed to `std::array<uint64_t, 4>` then this is not needed.